### PR TITLE
Enable code coverage when "Code Coverage;arg1=val1;arg2=val2" is provided in cli

### PR DESCRIPTION
**Pull request by fhnaseer**
This will enable code coverage for the given command and will also append testAdapterPath to child dotnet process.

dotnet test --collect:"Code Coverage;arg1=val1"

will call

"dotnet" exec vstest.console.dll D:\TestProject\bin\Debug\net6.0\TestProject.dll --collect:"Code Coverage;arg1=val1" --testAdapterPath:D:\NuGetFeed\microsoft.codecoverage\16.9.4\build\netstandard1.0\

### CODE REVIEW

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs [hunk 1]
```diff
@@ -368,7 +368,12 @@ private List<string> AddArgs()
             {
                 foreach (var arg in this.VSTestCollect)
                 {
-                    if (arg.Equals("Code Coverage", StringComparison.OrdinalIgnoreCase))
+                    // Argument value could be CodeCoverage;a=b;c=d.
+                    // Split arg with ; and compare with the first token.
+                    var tokens = arg.Split(';');
+
+                    if (arg.Equals("Code Coverage", StringComparison.OrdinalIgnoreCase) ||
```
Comment (MarcoRossignoli) [inline src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs:375]: `StartWith` does work?

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs [hunk 2]
```diff
@@ -368,7 +368,12 @@ private List<string> AddArgs()
             {
                 foreach (var arg in this.VSTestCollect)
                 {
-                    if (arg.Equals("Code Coverage", StringComparison.OrdinalIgnoreCase))
+                    // Argument value could be CodeCoverage;a=b;c=d.
```
Comment (MarcoRossignoli) [inline src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs:371]:
```suggestion
                    // Argument value could be "Code Coverage;a=b;c=d".
```

**Review by Sanan07 — APPROVED** [review]

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs (hunk 1 above)
Comment (Sanan07) [inline src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs:375]: Maybe better to have const string with value = "Code Coverage" for correctness. If someone changes the value in future we will not get misspelling.

**Review by MarcoRossignoli — APPROVED** [review]
LGTM

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs (hunk 1 above)
Comment (fhnaseer) [reply src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs]: Updated

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs [hunk 3]
```diff
@@ -20,6 +20,7 @@ public class VSTestTask : Task, ICancelableTask
         private VSTestForwardingApp vsTestForwardingApp;
 
         private const string vsTestAppName = "vstest.console.dll";
+        private const string CodeCovergaeString = "Code Coverage";
```
Comment (Sanan07) [inline src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs:23]: nit : misspelling
```suggestion
        private const string CodeCoverageString= "Code Coverage";
```

**Review by jakubch1 — APPROVED** [review]

**Pull request merged**